pkg/distributor/distributor.go

```diff
@@ -1223,9 +1223,9 @@ func (d *Distributor) send(ctx context.Context, ingester ring.InstanceDesc, time
 	_, err = c.Push(ctx, &req)
 	if resp, ok := httpgrpc.HTTPResponseFromError(err); ok {
 		// Wrap HTTP gRPC error with more explanatory message.
-		return httpgrpc.Errorf(int(resp.Code), "failed pushing to ingester: %s", resp.Body)
+		return httpgrpc.Errorf(int(resp.Code), "failed pushing to ingester: address=%s body=%s", ingester.Addr, resp.Body)
```

Comment:
This error will be exposed to the client. We're already exposing internal IPs, but I'm wondering if exposing the body may contain sensitive info (none that comes to my mind right now, to be honest).

Reply:
I was also wondering if this might expose the address to the client, and whether it's desirable :)

Reply:
hmm, true. There's two ways to deal with this: keep the error and log it, but send a sanitized/generic version to the caller. Simplify the error message and log the details directly in the send function. Not sure which one we'd usually do.

Reply:
Scanning through https://github.com/grafana/mimir/tree/main/docs/internal/contributing there's no guidance on this.

Reply:
Idea: make sure the info is in a trace span, and make sure the trace ID can be found by the person investigating.